server/lib/report_server/portal_dbs.ex

```diff
@@ -42,10 +42,20 @@ defmodule ReportServer.PortalDbs do
     end)
   end
 
+  defp value_sorter(v1, v2) do
```

Comment:
You could replace this with something like this to pattern match on the stuct type of the params:

```
  defp value_sorter(v1 = %Date{}, v2 = %Date{}), do: Date.compare(v1, v2) != :gt
  defp value_sorter(v1, v2), do: v1 <= v2
```